### PR TITLE
Fix KennaUpdateAsset

### PR DESCRIPTION
**Pull request by DaniSalcedoGFT**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
We moved the command "kenna-update-asset" from the commands that works with return-result to the commands that works with callable, because the result of the command didn't appear in the context data when we add the argument extend-context.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by DaniSalcedoGFT** [comment]
Hi @anas-yousef, sorry to bother but there is something missing? When is expected to be reviewed this topic?

File: Packs/Kenna/Integrations/KennaV2/KennaV2.py [hunk 1]
```diff
@@ -1,5 +1,3 @@
-import demistomock as demisto  # noqa: F401
-from CommonServerPython import *  # noqa: F401
```
Comment (anas-yousef) [inline Packs/Kenna/Integrations/KennaV2/KennaV2.py:2]: These are necessary, deleting them will break the script.

**Review by anas-yousef — COMMENTED** [review]
Hey @DaniSalcedoGFT, thank you so much for your contribution!
Can you please explain what you mean with this sentence?
**because the result of the command didn't appear in the context data when we add the argument extend-context.**
I can see that in the old version, we only returned a string to the War Room whether the request succeeded or not, without return anything to the context data.
I believe your changes prove the same behavior. If you want we can have a zoom meeting regarding this, so I can understand your intuition better :)

**Comment by DaniSalcedoGFT** [comment]
Hi @anas-yousef, in the old version as you say it didn't show anything in the context-data, with this new version it won't show it anything on the context-data also but with the old version when we added the extend-context argument the result showed as null, this change resolves that so it shows correctly.
With the previous sentence I explained wrong mention the context-data, sorry.
Example, we execute this command, with the old version the argument extend-context it didn't work, with the new changes now show us the result of the command.
!kenna-update-asset id="1280160" notes="Asset" extend-context="assetKenna="

**Comment by anas-yousef** [comment]
@DaniSalcedoGFT I didn't quite get the difference. Could we have a zoom call regarding this?
Are you available over [DFIR](https://start.paloaltonetworks.com/join-our-slack-community) slack channel?

**Comment by DaniSalcedoGFT** [comment]
Hi, I am so short of time to make calls so I will try to explain it again with some images.
We use the command **!kenna-update-asset id="1280160" notes="Asset" extend-context="assetKenna="**
With the current version of the integration we see that the in the Context Data appears the assetKenna but with NULL value
Kenna_1

With the changes we are pushing we achieve that in the Context Data appear the right values
Kenna_2

I hope this clear the doubts you have and we can push this change the sooner we can, if this don't clear something i'm not available on DFIR slack channel so don't know where we could do something.

**Comment by anas-yousef** [comment]
@DaniSalcedoGFT Thank you for your explanation, I will go over this with my manager and see why this happens.
The PR review might be slightly delayed due to the upcoming holiday, I apologize in advance. The holiday will be over by April 30th.

**Comment by DaniSalcedoGFT** [comment]
Sure, let us know when there is some new information about this topic

File: Packs/Kenna/Integrations/KennaV2/KennaV2.py [hunk 2]
```diff
@@ -399,8 +399,8 @@ def update_asset_command(client: Client, args: dict[str, str]) -> CommandResults
 
     result = client.http_request(message='PUT', suffix=url_suffix, data=asset)
     if result.get('status') != "success":
-        return CommandResults(readable_output=f'Could not update asset with ID {asset_id}.', raw_response=result)
-    return CommandResults(readable_output=f'Asset with ID {asset_id} was successfully updated.')
+        return f'Could not update asset with ID {asset_id}.',{},[]
+    return f'Asset with ID {asset_id} was successfully updated.',{},[]
```
Comment (anas-yousef) [inline Packs/Kenna/Integrations/KennaV2/KennaV2.py:403]:
```suggestion
        return CommandResults(readable_output=f'Could not update asset with ID {asset_id}.', raw_response=result)
    return CommandResults(readable_output=f'Asset with ID {asset_id} was successfully updated.', raw_response=result)
```
If you supply the **raw_response=result** to CommandResults, you should get the behavior that you want.
Don't forget to use **return_results** in the main function.

File: Packs/Kenna/Integrations/KennaV2/KennaV2.py [hunk 3]
```diff
@@ -399,8 +399,8 @@ def update_asset_command(client: Client, args: dict[str, str]) -> CommandResults
 
     result = client.http_request(message='PUT', suffix=url_suffix, data=asset)
```
Comment (anas-yousef) [inline Packs/Kenna/Integrations/KennaV2/KennaV2.py:400]:
```suggestion
    result = client.http_request(message='PUT', suffix=url_suffix, data=asset)
    readable_output = f'Asset with ID {asset_id} was successfully updated.'
```

File: Packs/Kenna/Integrations/KennaV2/KennaV2.py [hunk 4]
```diff
@@ -399,8 +399,8 @@ def update_asset_command(client: Client, args: dict[str, str]) -> CommandResults
...
     if result.get('status') != "success":
```
Comment (anas-yousef) [inline Packs/Kenna/Integrations/KennaV2/KennaV2.py:401]:
```suggestion
    if result.get('status') != "success":
        readable_output = f'Could not update asset with ID {asset_id}.'
```

File: Packs/Kenna/Integrations/KennaV2/KennaV2.py (hunk 2 above)
Comment (anas-yousef) [inline Packs/Kenna/Integrations/KennaV2/KennaV2.py:403]: In order to get the same behavior as before using **CommandResults**, you should do the following:
```suggestion
    return CommandResults(readable_output= readable_output, raw_response=readable_output)
```

**Review by anas-yousef — COMMENTED** [review]
@DaniSalcedoGFT Please review my comments.
Don't forget to change to **return_results** in the main function.

**Comment by DaniSalcedoGFT** [comment]
Hi @anas-yousef, we will try with your changes and come back with results

**Comment by anas-yousef** [comment]
Hey @DaniSalcedoGFT, any updates?

**Comment by anas-yousef** [comment]
Hi @DaniSalcedoGFT,
Thank you for your contribution. 
I have not received your response in the last two weeks, so I will close your PR now.
Please feel free to re-open it whenever you're available. 
Thanks again.

**Pull request closed without merging**